### ISSUE TITLE
YouTube audio download failure due to running out of space or battery not handled gracefully

### ISSUE THREAD

**Issue by galfisk** [TeamNewPipe/NewPipe#10485]
### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.25.2

### Steps to reproduce the bug

1. Download audio from any YouTube video, .m4a format (haven't tested with video downloads, or with other audio formats)
2. Let the download run in the background (haven't tested watching the download in the app)
3. Run out of battery or storage space while downloading

### Expected behavior

I expected NewPipe to either automatically retry or resume the download, or to let me do that manually.

### Actual behavior

NewPipe thinks that the download completed successfully, and there's no option to retry or resume. However, the file itself is not complete.

### Affected Android/Custom ROM version

Android 10

### Affected device model

Samsung Galaxy A6 SM-A600FN/DS

**Comment by galfisk** [comment]
Yes, kind of: in the case of running out of space, I get a warning
notification that the download failed, but when I go to the NewPipe
download manager, the download keeps counting up and eventually reaches
100%.

Den tis 10 okt. 2023 16:23opusforlife2 ***@***.***> skrev:

**Comment by CloudyRowly** [comment]
Hi, can I have a go at investigating this issue? Thank you!

**Comment by CloudyRowly** [comment]
Can I have a go at investigating this issue? Thanks!

**Comment by CloudyRowly** [comment]
I believe I have came up with quite a solution that solves other issues related to full memory as well. Can I get this issue assigned to me so I can create a corresponding PR? Thanks
